taqueria-plugin-tezos-client/client.ts

```diff
@@ -0,0 +1,256 @@
+    sourcePath = await execCmd(getCheckFileExistenceCommand(sandbox, sourcePath))
+                       .then(() => sourcePath)
+                       .catch(() => getInputFilenameFromContractDir(opts, sourceFile))
+
+    const simulateContractHelper = () => {
+        try {
+            const cmd = getSimulateCommand(opts, sandbox, sourceFile, sourcePath)
+            return execCmd(cmd)
+            .then(async ({stdout, stderr}) => { // How should we output warnings?
+                if (stderr.length > 0) sendErr(`\n${stderr}`)
+                return {
+                    contract: sourceFile,
+                    result: stdout
+                }
+            })
+            .catch(err => {
+                sendErr(" ")
+                sendErr(err.message.split("\n").slice(1).join("\n"))
+                return Promise.resolve({
+                    contract: sourceFile,
+                    result: "Invalid"
+                })
+            })
+        } catch (err) {
+            sendErr(" ")
+            sendErr((err as Error).message)
+            return Promise.resolve({
+                contract: sourceFile,
+                result: "Bad input or storage value"
+            })
+        }
+    }
+
+    return execCmd(getCheckFileExistenceCommand(sandbox, sourcePath))
+    .then(simulateContractHelper)
+    .catch(err => {
+        sendErr(" ")
+        sendErr(sourceFile + ": Does not exist\n")
+        return Promise.resolve({
+            contract: sourceFile,
+            result: "N/A"
+        })
+    })
+}
+
+const simulate = <T>(parsedArgs: Opts, sandbox: Sandbox): LikeAPromise<PluginResponse, Failure<T>> => {
+    if (parsedArgs.sourceFile) {
+        return simulateContract (parsedArgs, sandbox) (parsedArgs.sourceFile as string)
+               .then(data => [data])
+               .then(sendAsyncJsonRes)
+    }
+    return sendAsyncErr(`Please specify a contract. E.g. taq simulate local <contractName> ...`)
+}
+
+const simulateTask = async <T>(parsedArgs: Opts) : LikeAPromise<void, Failure<T>> => {
+    const sandbox = getSandbox(parsedArgs)
+    if (sandbox) {
+        if (doesUseFlextesa(sandbox)) {
+            return await isSandboxRunning(sandbox.name.value)
+            ? simulate(parsedArgs, sandbox).then(sendJsonRes)
+            : sendAsyncErr(`The ${sandbox.name} sandbox is not running.`)
+        }
+        return sendAsyncErr(`Cannot start ${sandbox.label} as its configured to use the ${sandbox.plugin} plugin.`)
+    }
+    return sendAsyncErr(`There is no sandbox configuration with the name ${parsedArgs.sandboxName}.`)
+}
+
+export const client = <T>(parsedArgs: Opts) : LikeAPromise<void, Failure<T>> => {
+    switch (parsedArgs.task) {
+        case 'typecheck':
+            return typecheckTask(parsedArgs)
+        case 'simulate':
+            return simulateTask(parsedArgs)
+        default:
+            return sendAsyncErr(`${parsedArgs.task} is not an understood task by the Tezos-client plugin`,)
```

Comment:
This messaging should be improved

At a minimum:
```
${parsedArgs.task} is not a valid Taqueria task
```

Ideally something actionable like:
```
${parsedArgs.task} is not a valid task. Tezos-client plugin tasks are: /n
taq typecheck
taq simulate
```

Reply:
Taking a second look, I think this should never happen. That would be a mismatch between `index.ts` and this case statement.